apis/python/src/tiledbsoma/experiment_query/axis.py

```diff
@@ -0,0 +1,68 @@
+from dataclasses import dataclass
+from typing import Optional, Tuple, Union
+
+import numpy as np
+import numpy.typing as npt
+import pyarrow as pa
+
+AxisCoordinate = Union[slice, int, npt.ArrayLike]
+AxisCoordinates = Tuple[AxisCoordinate, ...]
+AxisValueFilter = str
+
+
+@dataclass()
```

Comment:
in somacore we use `attrs`, which offers some niceties that might be useful here:

- `@some_attr.validator` adds a check that is run after initialization
- [converters](https://www.attrs.org/en/stable/init.html#converters) to transform the attribute into the form you want it

Reply:
We use `dataclasses` elsewhere in the package, which is why I went this direction.  If our intent  is to move to `attrs`, I'm happy to make this change.

Can I take the suggestion as indication we are migrating (over time of course)?

Reply:
I converted to attrs.

Please note: I also simplified this by removing all conversion, and delegating it to `DataClass` where it belongs.  This removes some nice features (eg, automatic conversion of `numpy.typing.ArrayLike`), but I filed issues to request that those be added to the DataFrame & NDArray classes where they belong (#688)